### ISSUE TITLE
New field: open hours

### ISSUE THREAD

**Issue by Lyquix** [FLEXIcontent/flexicontent-cck#358]
Hello guys - a few years ago we developed a custom field for entering open hours for business locations. This was developed for J1.5 and now we are upgrading the site to J3. We will be upgrading this field and I wanted to know if you are interested in making it part of the FC official core.

This is how the edit form looks like:

![](http://i.imgur.com/Z4G7vcf.png)

and this is how it renders in the frontend:

![](http://i.imgur.com/TnJrCrd.png)

or like this for individual days:

![](http://i.imgur.com/ZfvY8RE.png)

Here are some of the options that will be available:

* 12 or 24-hour options
* Option to hide minutes for o'clock hour
* Full day name (Monday) or 3-letter day name (Mon)
* Hours drop down list intervals: 5min, 10min, 15min, 20min, 30min, 60min
* Week start day: Monday or Sunday
* Improved UI for editing form: 
  * Select one day (Mon), multiple days (Mon, Wed, Thu), or ranges of days (Mon-Wed)
  * Select one range of hours (9am - 5pm), or multiple ranges (9am-12pm, 2pm-6pm)
* Additional free text field

**Comment by ggppdk** [comment]
yes, surely, this sound like a very useful field for some particular websites

**Comment by Lyquix** [comment]
Ok, perfect, we will work on it and send it as a pull request when we have it ready

**Comment by micker** [comment]
great thanks !

**Comment by micker** [comment]
think if it possible 
to add multivalue option => after we can use it in groupfield

**Comment by Lyquix** [comment]
Sure, I will let you know if we have questions about that

**Comment by Lyquix** [comment]
Hello guys! The plugin is ready for you to test. You can view the code and download from here: https://github.com/Lyquix/plg_fc_businesshours. It should be pretty self explanatory.

A few notes:

* **Use in group:** we added the code for supporting using in group, but we haven't tested it, and I am not very experienced with field groups, so you may need to take a close look to the code.
* **Multi-value:** we build the field to be multi value always. One reason is that we thought that if we need to limit the number of entries, we can set max_values, the other (more important) is that we need more work to de-couple the multi-value js of individual hours-rows from the multi-value js for days-hours rows (you will understand once you see it)

As I mentioned before, please take this to be part of the FLEXIcontent core plugins. I placed this in a separate repo because I wanted you to test it first.

**Comment by Lyquix** [comment]
@micker @ggppdk : did you have a chance to test this field? do you have any feedback?

**Comment by micker** [comment]
i will do it this evening promis
did you have time to test this => https://github.com/micker/mod_flexigooglemap

**Comment by micker** [comment]
hello its realy interesting
=> in backend button to delet time is too proche with delet move button
=> i think you can use joomla date function to have an automatic format and translation for output
=> maybe you can add title (simple text field) because when we use it in multi value mode it will be more simply than group. Ex: home hours, buissness hours, support hours ect
good field

**Comment by ggppdk** [comment]
The plugin name is good, about testing, lately i was more in bug-fixing (see my recent commits) and into private work. I did not have time to test this yet.

**Comment by ggppdk** [comment]
@Lyquix 

Can we still intergate this field to the core package ?
If yes will do in next major release

**Comment by lyquix-owner** [comment]
Yes, go for it, you can find it here: https://github.com/Lyquix/plg_fc_businesshours

**Comment by micker** [comment]
good idea !

**Comment by ggppdk** [comment]
We will be adding this field

and a few other fields
- qrcode field
- local media player field
- currency field
